main/activity-log.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+const Store = require('electron-store')
+const configStore = new Store({
+  name: 'activity-log'
+})
+
+class ActivityLog {
+  #entries
+  #lastId
+
+  constructor () {
+    this.#entries = loadStoredEntries()
+    this.#lastId = this.#entries
+      // We are storing ids as strings to allow us to switch to GUIDs in the future if needed
+      // When looking for the max id used, we need to cast strings to numbers.
+      .map(e => +e.id)
+      // Find the maximum id or return 0 when no events were recorded yet
+      .reduce((max, it) => it > max ? it : max, 0)
```

Comment:
```suggestion
    // We are storing ids as strings to allow us to switch to GUIDs in the future if needed
    // When looking for the max id used, we need to cast strings to numbers.
    // Find the maximum id or return 0 when no events were recorded yet
    this.#lastId = Math.max(...[0, this.#entries.map(e => Number(e.id))])
```

or

```suggestion
    this.#lastId = Math.max(...[
      // We are storing ids as strings to allow us to switch to GUIDs in the future if needed
      // When looking for the max id used, we need to cast strings to numbers.
      this.#entries.map(e => Number(e.id)),
      // Find the maximum id or return 0 when no events were recorded yet
      0
    ])
```

To me `Math.max()` is easier to reason about than `Array#reduce`. What do you think?

Reply:
Agreed, I'll use the second suggestion 👍🏻